src/Core/Services/Fido2AuthenticatorService.cs

```diff
@@ -36,28 +36,42 @@ public async Task<Fido2AuthenticatorMakeCredentialResult> MakeCredentialAsync(Fi
                 throw new NotSupportedError();
             }
 
-            await userInterface.EnsureUnlockedVaultAsync();
-            await _syncService.FullSyncAsync(false);
+            string cipherId = null;
+            var userVerified = false;
+            var accountSwitched = false;
 
-            var existingCipherIds = await FindExcludedCredentialsAsync(
-                makeCredentialParams.ExcludeCredentialDescriptorList
-            );
-            if (existingCipherIds.Length > 0)
+            do
             {
-                await userInterface.InformExcludedCredentialAsync(existingCipherIds);
-                throw new NotAllowedError();
-            }
+                try
+                {
+                    accountSwitched = false;
 
-            var response = await userInterface.ConfirmNewCredentialAsync(new Fido2ConfirmNewCredentialParams
-            {
-                CredentialName = makeCredentialParams.RpEntity.Name,
-                UserName = makeCredentialParams.UserEntity.Name,
-                UserVerificationPreference = makeCredentialParams.UserVerificationPreference,
-                RpId = makeCredentialParams.RpEntity.Id
-            });
+                    await userInterface.EnsureUnlockedVaultAsync();
+                    await _syncService.FullSyncAsync(false);
+
+                    var existingCipherIds = await FindExcludedCredentialsAsync(
+                        makeCredentialParams.ExcludeCredentialDescriptorList
+                    );
+                    if (existingCipherIds.Length > 0)
+                    {
+                        await userInterface.InformExcludedCredentialAsync(existingCipherIds);
+                        throw new NotAllowedError();
+                    }
+
+                    (cipherId, userVerified) = await userInterface.ConfirmNewCredentialAsync(new Fido2ConfirmNewCredentialParams
+                    {
+                        CredentialName = makeCredentialParams.RpEntity.Name,
+                        UserName = makeCredentialParams.UserEntity.Name,
+                        UserVerificationPreference = makeCredentialParams.UserVerificationPreference,
+                        RpId = makeCredentialParams.RpEntity.Id
+                    });
+                }
+                catch (AccountSwitchedException)
+                {
+                    accountSwitched = true;
+                }
+            } while (accountSwitched);
```

Comment:
Let's move forward for now but revisit this later. Created [PM-7140 ](https://bitwarden.atlassian.net/browse/PM-7140)